### ISSUE TITLE
Polar pattern breaks the body

### ISSUE THREAD

**Issue by knipknap** [FreeCAD/FreeCAD#15880]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

In the attached file, the Polar pattern no longer works.

![image](https://github.com/user-attachments/assets/036f7fb3-e4ca-4c67-9723-8c5a072dc565)

The same file works fine in 0.22.0dev.37898.

[broken-hub.zip](https://github.com/user-attachments/files/16611186/broken-hub.zip)

### Full version info

```shell
OS: Ubuntu 22.04.4 LTS (ubuntu:GNOME/ubuntu-xorg)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38459 (Git) AppImage
Build type: Release
Branch: main
Hash: 58729b3ab0060a1525fc45c87beabfd4b3d5e22e
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * DynamicData 2.62.0
  * btl 0.9.9
  * WebTools
  * toSketch
  * fasteners 0.4.62
  * lattice2 1.0.0
  * freecad.gears 1.2.0
  * sheetmetal 0.4.23
  * CycloidGearBox
  * parts_library
  * Assembly4 0.12.3
  * btl.backup1722529615.8976307 0.9.9 (Disabled)
```

### Subproject(s) affected?

None

### Anything else?

While I found some other existing issues regarding Polar Pattern (e.g. #15018, #14769), none of them seem obviously related.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luxwarp** [comment]
I checked your file with latest versions and commits that adresses the pattern problems and your file workes correctly so this is fixed in #15869

**Comment by knipknap** [comment]
Great, Thanks a lot for the fast response! I'll wait for the next weekly then.
